src/services/utility-helpers.js

```diff
@@ -768,3 +768,122 @@ export const groupBy = (arry, key) => {
 export const sortAnnotations = (annotations) => {
   return annotations.sort((a, b) => a.time?.start - b.time?.start);
 };
+
+/**
+ * Truncates text that may contain HTML to a given maximum character length
+ * while preserving the HTML structure
+ * 
+ * @param {String} htmlString string to truncate which might contain HTML markup
+ * @param {Number} maxLength allowed max character length
+ * @returns {Object} { truncated: String, isTruncated: Boolean }
+ */
+export const truncateText = (htmlString, maxLength) => {
+  const ellipsis = '...';
+  if (htmlString.length <= maxLength) {
+    return { truncated: htmlString, isTruncated: false };
+  }
+
+  // Create a temporary div to work with the HTML
+  const tempDiv = document.createElement('div');
+  tempDiv.innerHTML = htmlString;
+
+  const textLength = getTextLength(tempDiv);
+
+  // Add length of ellipsis (3) towards maxLength when truncating
+  if (textLength <= maxLength + ellipsis.length) {
+    // If the text content within HTML is shorter than maxLength return the original
+    return { truncated: htmlString, isTruncated: false };
+  } else {
+    // Truncate text only nodes
+    if (maxLength > 0) {
+      truncateNode(tempDiv, maxLength);
+    }
+    // Add ellipsis to the last text node
+    const lastTextNode = findLastTextNode(tempDiv);
+    if (lastTextNode) {
+      lastTextNode.textContent += ellipsis;
+    }
+    return { truncated: tempDiv.innerHTML, isTruncated: true };
+  }
+};
+
+/**
+ * Get the length of text within the given string that might contain HTML
+ * @param {Node} node node with text content
+ * @returns {Number} length of text content
+ */
+const getTextLength = (node) => {
+  if (node.nodeType === Node.TEXT_NODE) {
+    return node.textContent.length;
+  }
+  let length = 0;
+  for (const childNode of node.childNodes) {
+    length += getTextLength(childNode);
+  }
+  return length;
+};
+
+/**
+ * Truncate text content avoiding the HTML tags
+ * @param {Node} node node to truncate
+ * @param {Number} maxLength 
+ * @returns {Number} number of used characters by the current node
+ */
+const truncateNode = (node, maxLength) => {
+  // Truncate text nodes
+  if (node.nodeType === Node.TEXT_NODE) {
+    if (node.textContent.trim() === '') return 0;
+    if (node.textContent.length <= maxLength) {
+      return node.textContent.length;
+    } else {
+      // Get the index of the last space character in the truncated text with maxLength
+      const lastSpaceIndex = node.textContent.substring(0, maxLength).lastIndexOf(' ');
+      // If text doesn't have spaces, truncated at maxLength (this is probably an edge case?)
+      // FIXME:: Maybe there's a better way to handle this than breaking the word?
```

Comment:
If the node would be truncated and there is no space, should this truncate the entire textual content of the node?  That might work for English but probably not for Chinese or Japanese and other languages without required spaces.

For example, the last test in utility-helpers.test.js has to truncate mid-word because it only has one word in the node.

What's currently in this PR is probably good enough for now since it improves upon what was there before.  :slightly_smiling_face: